crates/pallets/starknet/src/tests/utils.rs

```diff
@@ -112,3 +116,58 @@ pub fn build_get_balance_contract_call(account_address: StarkFelt) -> (EntryPoin
 
     (balance_of_selector, calldata)
 }
+
+pub fn create_resource_bounds() -> starknet_api::transaction::ResourceBoundsMapping {
+    let mut map = BTreeMap::new();
+    map.insert(
+        starknet_api::transaction::Resource::L1Gas,
+        starknet_api::transaction::ResourceBounds { max_amount: 10000, max_price_per_unit: 12000 },
+    );
+    map.insert(
+        starknet_api::transaction::Resource::L1Gas,
+        starknet_api::transaction::ResourceBounds { max_amount: 50000, max_price_per_unit: 31000 },
+    );
+    starknet_api::transaction::ResourceBoundsMapping(map)
+}
+
+fn stark_felt_to_field_element(felt: &StarkFelt) -> FieldElement {
+    // Can only fail if the 32 byte number is not smaller than the field's modulus,
+    // but StarkFelt already enforces that
+    FieldElement::from_bytes_be(&felt.0).unwrap()
+}
+
+pub fn get_balance_contract_call(
+    sender_address: ContractAddress,
+    contract_address: ContractAddress,
+) -> Vec<Felt252Wrapper> {
+    let call_args = build_get_balance_contract_call(sender_address.0.0);
+    Starknet::call_contract(contract_address, call_args.0, call_args.1).unwrap()
+}
```

Comment:
It is just used one in one test, no?
Just write the call there and don't factorize yet

Reply:
`get_balance_contract_call` is used 10 times in `invoke_tx.rs`.

Reply:
okay fine then